src/nnpack/interface.jl

```diff
@@ -0,0 +1,25 @@
+include("impl.jl")
+
+
+for (front_name, backend) in (
+        :conv          => :_nnpack,
+        :∇conv_data    => :_nnpack,
+        :∇conv_filter  => :_nnpack,
+    )
+    @eval begin
+        @timeit_debug to function $(Symbol("$(front_name)$(backend)!"))(
+                        out::Array{T1,4}, in1::Array{T2,4}, in2::Array{T3,4},
+                        cdims::ConvDims; kwargs...) where {T1, T2, T3}
+            # Output must of the same type as in the function signature
+            T1.($(Symbol("$(front_name)$(backend)!"))(Float32.(out), Float32.(in1),
+                                                      Float32.(in2), cdims; kwargs...))
```

Comment:
Let's create two methods here; one that is specialized on `T1`, `T2`, `T3` all equal to `Float32`, and then one that is generic.  Let's include a `@warn "Automatically converting $(size(in1)) input tensor to Float32" maxlog=1` in the generic version so that users are aware that this is sub-optimal and can manually convert when necessary.

Reply:
if all of them are equal to Float32 won't it be exactly same as the function defined in nnpack/impl.jl

Reply:
You're right; you've already got the base case defined, so just put a `@warn` in this one.